packages/tools/viewer-alpha/src/viewer.ts

```diff
@@ -929,6 +890,48 @@ export class Viewer implements IDisposable {
         return true;
     }
 
+    private _suspendRendering(): IDisposable {
+        this._renderLoopController?.dispose();
+        this._suspendRenderCount++;
+        let disposed = false;
+        return {
+            dispose: () => {
+                if (!disposed) {
+                    disposed = true;
+                    this._suspendRenderCount--;
+                    if (this._suspendRenderCount === 0) {
+                        this._beginRendering();
+                    }
+                }
+            },
+        };
+    }
+
+    private _beginRendering(): void {
+        if (!this._renderLoopController) {
+            const render = () => {
+                this._details.scene.render();
+                if (this.isAnimationPlaying) {
+                    this.onAnimationProgressChanged.notifyObservers();

```

Comment:
not sure to understand why the progress is triggered every frame without data ?

Reply:
I was more thinking of the notification frequency, would not this run everyframe ? which as long as it animates we would expect ?

Reply:
Oh, yes this is reporting "progress" of the animation. It's just intended to make it easy to track animation state for things like updating the animation timeline.

Reply:
is it not similar to simply using isAnimationPlaying in this case ?

Reply:
isAnimationPlaying is just state, as is animationProgress. The observable is so you know when the animation progress has changed. You could hook into scene.onAfterRenderObservable but from an API standpoint I'm trying to make it very simple and not require knowledge of the Babylon.js API.

Reply:
For example, this is where the viewerElement registers for the `onAnimationProgressChanged` observable and updates its own state to trigger the animation timeline to update: https://github.com/BabylonJS/Babylon.js/blob/master/packages/tools/viewer-alpha/src/viewerElement.ts#L982-L983